docs/en/contributing-to-appium/developers-overview.md

```diff
@@ -0,0 +1,150 @@
+
+For an overview of the package hierarchy and the role that each package plays,
+please check out our [package
+overview](/docs/en/contributing-to-appium/appium-packages.md) doc.
+
+### Transpilation
+
+Appium is written in a new form of JavaScript, called ES6 (or now ES2015).
+Because this version of the language is not yet supported natively by older
+versions of node.js, Appium code is _transpiled_ to ES5 (the more
+widely-supported version of JS). This transpilation process must occur before
+any code is run. In addition to the new language features of ES6, we have
+adopted two very important keywords from the _subsequent_ version of JS, namely
+`async` and `await`, which assist in writing asynchronous code cleanly. Because
+of the transpilation step, Appium packages include tools which watch code for
+changes and automatically re-transpile the code. Usually, the same tool will
+automatically run unit tests as well to ensure that nothing small has broken.
+Most Appium packages have this as the default behavior when running `gulp`.
+
+### Linting and Style
+
+It's important for all of Appium's JS to look and feel the same. This includes
+style conventions as well as coding patterns and which libraries we use to
+solve various problems. You should get familiar with our new [ES2015 Style
+Guide](/docs/en/contributing-to-appium/style-guide-2.0.md). When transpiling,
+Appium packages will automatically run JSHint or other lint tools and provide
+warning or error feedback if the code doesn't conform to our style. These tools
+are not necessarily exhaustive of the kinds of style issues we care about, so
+we may also mention style issues in our reviews. This isn't to be nit-picky but
+to have a clean, consistent, and readable codebase!
+
+### Submitting Code
+
+Getting your code into Appium is easy: just submit a PR to one of our repos and
+engage with the maintainers in the review process. We have a number of
+requirements for code submissions:
+
+* Follow the style of the surrounding code and our Style Guide
+* Atomic commits--one commit per logical change (i.e., make sure that commits don't need to come in a group in order for the program to work. It should work at any given commit). Usually this means one commit per PR. You'll want to get very familiar with `git rebase -i` and squashing!
+* No merge commits: always rebase on top of latest master (or whatever other branch you're asking to merge into) before submitting your PR.
+* Almost all changes should have tests. Bugfixes should at the least have unit tests that prove that the bug has been fixed. New features should have unit tests and in most cases e2e tests to prove that the feature actually works. We will be happy to walk you through the test creation process. Reading the surrounding test code is a good place to start. Our CI systems usually run test coverage statistics and we will likely not merge code that decreases test coverage.
+
```

Comment:
These instructions are kinda scary for someone not used to contributing to open source. A younger me would be scared away. Maybe add a nice friendly note here like "Don't worry if this seems like a lot, we will helpfully and patiently work with you to get your code into Appium"